packages/typescript-estree/src/ts-estree/ts-estree.ts

```diff
@@ -856,27 +990,20 @@ export interface LabeledStatement extends BaseNode {
   body: Statement;
 }
 
-export interface Literal extends LiteralBase {
-  type: AST_NODE_TYPES.Literal;
-}
```

Comment:
I split this into 5 separate literal types, one for each valid literal value types (`boolean | number | RegExp | string | null`).
I did this so it's a bit nicer when defining types using literals.

I.e. we know that member names can only ever be a `StringLiteral` or a `NumberLiteral`, but never a `BooleanLiteral`, `RegExpLiteral`, etc.